search-api/src/search_api/models/document_access_request.py

```diff
@@ -0,0 +1,156 @@
+# Copyright © 2022 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Table for storing document access request details."""
+from datetime import datetime
+from enum import Enum
+from http import HTTPStatus
+
+from sqlalchemy import inspect
+from sqlalchemy.ext.hybrid import hybrid_property
+from sqlalchemy.orm import backref
+
+from search_api.exceptions import BusinessException
+
+from .db import db
+
+
+class DocumentAccessRequest(db.Model):
+    """Used to hold the document access request information."""
+
+    class Status(str, Enum):
+        """Enum of the request statuses."""
+
+        CREATED = 'CREATED'
+        PAID = 'PAID'
+        ERROR = 'ERROR'
+        COMPLETED = 'COMPLETED'  # After all docs are generated and stored.
+
+    __versioned__ = {}
+    __tablename__ = 'document_access_request'
```

Comment:
We aren't using versioned table here, as we're not including continuum.

Reply:
Removed.